### PR TITLE
feat: be able to configure the plugin with a file

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
This enables a plugin to inject the configuration inside the plugin manager and be able to configure a custom script with a custom main file.

Fixes https://github.com/cln-reckless/reckless.rs/issues/3

N.B: the file format can change in the future for now we keep the YAML file to be more similar to Docker compose file, maybe this will slow down the learning curve

Signed-off-by: Vincenzo Palazzo <vincenzopalazzodev@gmail.com>

### CODE REVIEW

File: reckless_github/src/repository.rs [hunk 1]
```diff
@@ -36,9 +41,22 @@ impl Github {
 
     /// Index the repository to store information
     /// related to the plugins
-    pub fn index_repository(&mut self) {
+    pub async fn index_repository(&mut self) -> Result<(), RecklessError> {
         let repo_path = &self.url.path_string;
         let pattern = format!("{}/[!.]*/*", &repo_path);
+
+        // FIXME rewrite it in a way that is more clear that
```
Comment (swaptr) [inline reckless_github/src/repository.rs:48]: Noted, lets move on with the merge and we will fix this later.

**Review by swaptr — APPROVED** [review]
Great work! Fantastic stream. Thanks!

**Comment by swaptr** [comment]
Rename [this](https://github.com/cln-reckless/reckless.rs/blob/c2c947f8b7ab25bcf2dbeca2eb79e48f36460b64/reckeless.yml) file

**Comment by vincenzopalazzo** [comment]
>Rename [this](https://github.com/cln-reckless/reckless.rs/blob/c2c947f8b7ab25bcf2dbeca2eb79e48f36460b64/reckeless.yml) file to reckless.yml

Thanks!

**Pull request merged**